autogpt/utils.py

```diff
@@ -11,10 +11,64 @@
 except:
     pass
 
+from autogpt.config import Config
 
-def clean_input(prompt: str = ""):
+cfg = Config()
+
+
+def clean_chat_message_to_user(report: str):
+    for plugin in enumerate(cfg.plugins):
+        if not hasattr(plugin, "can_handle_report"):
+            continue
+        if not plugin.can_handle_report():
+            continue
+        plugin.send_message(report)
+
+
+def clean_input(prompt: str = "", talk=False):
     try:
-        return input(prompt)
+        if cfg.chat_messages_enabled:
+            for plugin in enumerate(cfg.plugins):
+                if not hasattr(plugin, "can_handle_user_input"):
+                    continue
+                if not plugin.can_handle_user_input():
+                    continue
+                plugin_response = plugin.user_input(prompt, talk)
+                if plugin_response in [
+                    "yes",
+                    "yeah",
+                    "yep",
+                    "yup",
+                    "y",
+                    "ok",
+                    "okay",
+                    "sure",
+                    "affirmative",
+                    "aye",
+                    "aye aye",
+                    "alright",
+                    "alrighty",
+                ]:
+                    return "y"
+                elif plugin_response in [
+                    "no",
+                    "nope",
+                    "n",
+                    "nah",
+                    "negative",
+                    "nay",
+                    "nay nay",
+                ]:
```

Comment:
I prefer to have this kind of extensive parsing/mapping in the plugin instead of in the core. Otherwise, reduce the list to the obvious options, and make sure to fix the casing of the `plugin_response` before matching.

Reply:
I think I will first reduce this then